KarhooUISDK/UIComponents/LoyaltyView/KarhooLoyaltyPresenter.swift

```diff
@@ -342,42 +367,61 @@ final class KarhooLoyaltyPresenter: LoyaltyPresenter {
         return currentMode == .burn ? viewModel.burnAmount : 0
     }
     
-    private func getSubtitleText() -> String {
+    private func getEarnText() -> String {
         switch currentMode {
         case .none:
             return ""
         case .burn:
-            return String(format: NSLocalizedString(UITexts.Loyalty.pointsBurnedForTrip, comment: ""), "\(viewModel?.burnAmount ?? 0)")
+            return viewModel?.canEarn ?? false ?
+            getLocalizedEarnPointsText(for: 0) :
+            ""
         case .earn:
-            if viewModel?.canEarn ?? false {
-                return String(format: NSLocalizedString(UITexts.Loyalty.pointsEarnedForTrip, comment: ""), "\(viewModel?.earnAmount ?? 0)")
-            } else {
-                return ""
-            }
+            return viewModel?.canEarn ?? false ?
+            getLocalizedEarnPointsText(for: viewModel?.earnAmount ?? 0) :
+            ""
+        }
+    }
+    
+    private func getLocalizedEarnPointsText(for amount: Int) -> String {
+        String(
+            format: NSLocalizedString(UITexts.Loyalty.pointsEarnedForTrip, comment: ""),
+            "\(amount)"
+        )
+    }
+    
+    private func getBurnText() -> String {
+        switch currentMode {
+        case .none, .earn:
+            return viewModel?.canBurn ?? false ? UITexts.Loyalty.burnOffSubtitle : ""
```

Comment:
Whoa, what a bunch of code! Break it and create a local constant, so another developer can follow the flow easily.